src/editor/mod.rs

```diff
@@ -0,0 +1,82 @@
+use std::path::Path;
+use std::fs::File;
+use std::io::Read;
+
+pub static JS: &'static [u8] = include_bytes!("editor.js");
+pub static ACE_JS: &'static [u8] = include_bytes!("ace.js");
+pub static MODE_RUST_JS: &'static [u8] = include_bytes!("mode-rust.js");
+pub static THEME_DAWN_JS: &'static [u8] = include_bytes!("theme-dawn.js");
+pub static THEME_TOMORROW_NIGHT_JS: &'static [u8] = include_bytes!("theme-tomorrow_night.js");
+
+/// Integration of a JavaScript editor for playpens.
+/// Uses the Ace editor: https://ace.c9.io/.
+/// The Ace editor itself, the mode, and the theme files are the
+/// generated minified no conflict versions.
+///
+/// The `Editor` struct should be used instead of the static variables because
+/// the `new()` method
+/// will look if the user has an editor directory in his source folder and use
+/// the users editor instead
```

Comment:
I mean, they can be replaced, sure, but mostly it's for organization since it's also mirroring where the resources are to begin with and mirroring the config (which would be output > html > editor in this case). I added a new directory because there are a bunch of naturally organized files that relate to the editor and only the editor...

You want me to dump all of those into the `theme` folder on top of everything that's already in there? Which also implies I'd need to merge all the code, as well... I don't understand what this buys. This, in my view, would make the organization considerably worse. Many of these files are already too long and the folders already contain too many files irrelevant to development. Nor does it express the optionality of the editor in contrast to the rest of the theme.

And consider how confusing it would be to have `tomorrow_night.js` and `tomorrow_night.css` right next to each other when one of them is used only by the playpen editor and the other is always used by the main book?